backend/src/services/user/user.service.ts

```diff
@@ -129,7 +131,13 @@ class User {
     }));
   }
 
-  public async delete(id: number): Promise<boolean> {
+  public async delete(userId: number, id: number): Promise<boolean> {
+    if (userId === id) {
+      throw new PermissionsError({
```

Comment:
```suggestion
      throw new UsersError({
```
error should be per serivice